### DISCUSSION TITLE
Upcoming Mma 13.3

### DISCUSSION THREAD

**Discussion by rnlg** [rnlg/LiteRed2#2] [Announcements]
Mathematica 13.3 will have `TypeOf` function in ``System` `` context. This is in conflict with `TypeOf` procedure of RNL/Types.m, which was used in other RNL/*.m and also in LiteRed2xxx.m. 
Therefore, we replace ``Types`TypeOf`` with ``Types`ExpressionType``.

Note that the bases  created with previous versions of LiteRed2 will cause problems in Mma >=13.3 and also will not work with recent versions of LiteRed2. This can be fixed by replacing "TypeOf" with "ExpressionType" in all basis files. E.g., in linux, it can be achieved with the command
$ **sed -i 's/TypeOf/ExpressionType/g' \*** 
issued in the directory of the basis.

**Comment by jodavies** [comment]
Hi Roman, just to mention that the files for 1.84 linked here https://inp.nsk.su/~lee/programs/LiteRed/ do not actually contain the `TypeOf`->`ExpressionType` fix (and also still have 1.83 as the version, though a 1.84 filename).

Thanks,
Josh.

**Reply by rnlg** [reply]
Hi, Josh,

Thanks for letting me know. I gave up supporting previous versions of  LiteRed. But have you tested that it works with patched Types.m etc? If so, I can at least replace these files with updated versions.

**Reply by jodavies** [reply]
Yes, it seems fine with v14.1. One also needs to make the same changes for Mint (to install this I only copy `Mint.m` and `RNL/Mint11.m`, and let it use LiteRed's supporting files).
